client/sections-middleware.js

```diff
@@ -0,0 +1,179 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import config from 'config';
+import page from 'page';
+import { find, groupBy, toPairs, partial, escapeRegExp } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { activateNextLayoutFocus } from 'state/ui/layout-focus/actions';
+import utils from '../server/bundler/utils';
+import * as LoadingError from 'layout/error';
+import * as controller from './controller/index.web';
+import { restoreLastSession } from 'lib/restore-last-path';
+import { hub as preloadHub } from 'sections-preload';
+import { switchCSS } from 'lib/i18n-utils/switch-locale';
+
+import sections from './sections';
+
+const sectionsWithCss = sections.map( section => ( {
+	...section,
+	...( section.css && {
+		css: {
+			id: section.css,
+			urls: utils.getCssUrls( section.css ),
+		},
+	} ),
+} ) );
+
+const _loadedSections = {};
+
+function maybeLoadCSS( sectionName ) {
+	//eslint-disable-line no-unused-vars
+	const section = find( sectionsWithCss, elem => elem.name === sectionName );
```

Comment:
while you're at it, you might like `lodash.matchesProperty()`

```js
const section = find( sectionsWithCss, matchesProperty( 'name', sectionName ) )
```

Reply:
i'm pretty sure i could even do:
```js
const section = find( secttionWithCss, { name: sectionName } )
```